Added filter to remove sourcemap files for the Webpack plugin

mhaagens opened the pull request.

thejameskyle commented:
Is there a reason source map files shouldn't be included?

mhaagens commented:
They're the only output when I run the plugin with sourcemaps enabled.
So getBundles only returns source map files. The other option is I have something seriously wrong in my config somewhere :sweat_smile:

Output without filtering;

```
{
  "components/Properties": {
    "id": "aqA1",
    "name": "./src/shared/components/Properties.js",
    "file": "js/0.app.2afe9166051089ee8702.js.map"
  },
  "symbol-observable": {
    "id": "2MIV",
    "name": "./node_modules/symbol-observable/index.js",
    "file": "js/app.a5268c7b834736581164.js.map"
  },
  ...
}
```

With filtering:
```
{
  "components/Properties": {
    "id": "aqA1",
    "name": "./src/shared/components/Properties.js",
    "file": "js/0.app.2afe9166051089ee8702.js"
  },
  "symbol-observable": {
    "id": "2MIV",
    "name": "./node_modules/symbol-observable/index.js",
    "file": "js/app.a5268c7b834736581164.js"
  },
  ...
}
```

bertho-zero commented:
In this case I think that each entry in the chunks file should be an array with all the modules concerned, the problem here is that the module gets overwritten (loop in loop). There is no reason to filter the sourcemaps files, which in my opinion are essential to debug.

mhaagens commented:
Removed the filtering and added the sourceMap as a property, might not be exactly what you suggested though. Hopefully someone with a little more skill than me can merge a PR for this, because right now I can't use React Loadable due to the source map issue.

thejameskyle commented:
> In this case I think that each entry in the chunks file should be an array with all the modules concerned

I agree, I don't like all the handling for file extensions being added in PRs right now, we should just change the webpack plugin to output:

```js
{
  [rawRequest: string]: Array<{ // << array for each file
    id: number, // << maybe delete this
    name: string | null,
    file: string,
  }>
}
```

Then update the `getBundles` method appropriately:

```js
function getBundles(manifest, moduleIds) {
  return moduleIds.reduce((bundles, moduleId) => {
    return bundles.concat(manifest[moduleId]);
  }, []);
}
```

thejameskyle commented:
Done https://github.com/thejameskyle/react-loadable/commit/78dcb02dc02d9a0be50a64c3522d6688eb6d7e71
